packages/gatsby-source-nacelle/gatsby-node.js

```diff
@@ -1,18 +1,18 @@
 const sourceNodes = require('./src/source-nodes');
 const typeDefs = require('./src/type-defs');
 const { createRemoteImageFileNode, cmsPreviewEnabled } = require('./src/utils');
-const { nacelleClient } = require('./src/services');
+const { nacelleClient: createNacelleClient } = require('./src/services');
 
 exports.pluginOptionsSchema = ({ Joi }) => {
   return Joi.object({
-    nacelleSpaceId: Joi.string()
-      .required()
-      .description('Space ID from the Nacelle Dashboard'),
-    nacelleGraphqlToken: Joi.string()
-      .required()
-      .description('GraphQL Token from the Nacelle Dashboard'),
-    contentfulPreviewSpaceId: Joi.string().description(
-      'Space ID from Contentful Dashboard settings'
+    nacelleSpaceId: Joi.string().description(
+      'Space ID from the Nacelle Dashboard'
+    ),
+    nacelleGraphqlToken: Joi.string().description(
+      'GraphQL Token from the Nacelle Dashboard'
+    ),
+    nacelleClient: Joi.optional().description(
+      'Instance of the `@nacelle/client-js-sdk` which you want to use to fetch your nacelle data'
     ),
```

Comment:
I originally tried to do some conditional logic with Joi so that either the spaceId & GraphQLtoken is required, or the client is required, but I kept getting errors.

Reply:
I think the documentation-driven approach in this PR is totally fine!